feature-libs/product-configurator/rulebased/core/facade/configurator-cart.service.ts

```diff
@@ -162,17 +169,24 @@ export class ConfiguratorCartService {
       .requireLoadedCart()
       .pipe(take(1))
       .subscribe((cartState) => {
-        const cartId = this.commonConfigUtilsService.getCartId(cartState.value);
-        const parameters: Configurator.UpdateConfigurationForCartEntryParameters = {
-          userId: this.commonConfigUtilsService.getUserId(cartState.value),
-          cartId: cartId,
-          cartEntryNumber: configuration.owner.id,
-          configuration: configuration,
-        };
+        this.commonConfigUtilsService
+          .getUserId(cartState.value)
+          .pipe(take(1))
+          .subscribe((id) => {
+            const cartId = this.commonConfigUtilsService.getCartId(
+              cartState.value
+            );
+            const parameters: Configurator.UpdateConfigurationForCartEntryParameters = {
+              userId: id,
+              cartId: cartId,
```

Comment:
In all other methods cartId is not a variable an will be filled directly from the method return value. Is there a reason to first create a const cartId?